src/imgui/Shortcuts.hh

```diff
@@ -0,0 +1,82 @@
+#ifndef IMGUI_SHORTCUTS_HH
+#define IMGUI_SHORTCUTS_HH
+
+#include "ImGuiUtils.hh"
+
+#include "strCat.hh"
+#include "imgui_stdlib.h"
+
+#include <functional>
+#include <optional>
+#include <string_view>
+#include <vector>
+
+namespace openmsx {
+
+class ImGuiManager;
+class GlobalSettings;
+class SettingsConfig;
+
+enum ShortcutIndex {
+	GOTO_MEMORY_ADDRESS = 0,
+	GOTO_DISASM_ADDRESS,
+	NUM_SHORTCUTS,
+};
+
+enum ShortcutType {
+	LOCAL,    // if ImGui widget has focus
+	GLOBAL,   // inside ImGui layer only
+};
+
+class Shortcuts final
+{
+public:
+	struct Data {
+		ShortcutIndex index;
```

Comment:
This `index` field should not be stored. It should always be the case that (in the various tables) the entry at the `i`th position has `index == i`.

The only exception is `std::vector<Shortcuts::Data>` in `SettingsParser`. That one should store a small wrapper struct (or a std::pair) that contains both the index and the data.